components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/bean/SSOAgentConfig.java

```diff
@@ -190,10 +218,87 @@ public KeyStore getKeyStore() throws org.wso2.carbon.identity.sso.agent.exceptio
     }
 
     public void setKeyStore(KeyStore keyStore) {
+
         this.keyStore = keyStore;
     }
+
     public void initConfig(Properties properties) throws SSOAgentException {
 
+        char[] key = null;
+        String decodedPassword;
+        boolean isReadFile = false;
+        Properties copyOfProperties = new Properties();
+        copyOfProperties.putAll(properties);
+        Path path;
+        URL url = getClass().getClassLoader().getResource(SSOAgentConstants.SSOAgentConfig.SECRETKEY_FILEPATH);
+        for (Map.Entry<Object, Object> entry : copyOfProperties.entrySet()) {
+            if (String.valueOf(entry.getValue()).startsWith("Enc:")) {
+                if (System.getProperty("secretkey") == null) {
+                    if (!isReadFile) {
+                        if (url != null) {
+                            Map<String, String> env = new HashMap<>();
+                            try {
+                                URI uri = url.toURI();
+                                FileSystems.newFileSystem(uri, env);
+                                path = Paths.get(uri);
+                                byte[] filearray = Files.readAllBytes(path);
+                                key = new char[filearray.length];
+                                int i = 0;
+                                for (byte b : filearray) {
+                                    key[i] = (((char) filearray[i]));
+                                    i++;
+                                }
+                                isReadFile = true;
+                                if (Files.deleteIfExists(path)) {
+                                    LOGGER.info("Successfully deleted the file " + path);
+                                } else {
+                                    LOGGER.info("Can't find the file " + path + " to delete");
+                                }
+                            } catch (URISyntaxException e) {
+                                LOGGER.log(Level.SEVERE, "Error while converting from URL to URI ");
```

Comment:
Why just log ? Either throw it or if you are handling it better to print stacktrace as an error log.